drgn_tools/crash_net.py

```diff
@@ -0,0 +1,372 @@
+                    name,
+                    ips,
+                ]
+            )
+    print_table(rows)
+
+
+def crash_a(prog: Program) -> None:
+    """
+    Displays the arp cache.
+    :param prog: drgn program
+    :returns: None.
+    """
+
+    rows = [
+        ["NEIGHBOUR", "IP ADDRESS", "HW TYPE", "HW ADDRESS", "DEVICE", "STATE"]
+    ]
+
+    nht = prog["arp_tbl"].nht
+
+    hash_shift = nht.hash_shift.value_()
+    nhash_buckets = 1 << hash_shift
+    hash_buckets = nht.hash_buckets
+
+    for i in range(nhash_buckets):
+        neigh = hash_buckets[i]
+        while neigh:
+            neigh_addr = hex(neigh.value_())[2:]
+            ip_addr = ""
+            for j in range(4):
+                ip_addr += str(neigh.primary_key[j].value_()) + "."
+            ip_addr = ip_addr[:-1]
+
+            hw = neigh.ha
+            hw_addr = ""
+            for j in range(6):
+                hw_addr += str(hex(hw[j])[2:].zfill(2)) + ":"
+            hw_addr = hw_addr[:-1]
+            dev_name = neigh.dev.name.string_().decode()
+
+            hw_type = str(hwtype_dict.get(int(neigh.dev.type)))
+            if int(neigh.dev.type) not in hwtype_dict.keys():
+                hw_type = "ARPHRD_UNKNOWN"
+            hw_type = hw_type[7:]
+            dev_state = str(state_dict.get(int(neigh.nud_state.value_())))[4:]
+            rows.append(
+                [
+                    neigh_addr,
+                    ip_addr,
+                    hw_type,
+                    hw_addr,
+                    dev_name,
+                    dev_state,
+                ]
+            )
+            neigh = neigh.next
+    print_table(rows)
+
+
+def print_task_sockets(prog: Program, pid: int, print_full_data: bool) -> None:
+    """
+    Displays the socket addreses, family type, source and destination
+    addresses and corresponding ports along with the dump of data
+    structures. If print_full_data is true, it prints detailed data
+    structure dumps.
+    :param prog: drgn program
+    :param pid: pid of the process.
+    :returns: None.
+    """
+
+    task = find_task(prog, int(pid))
+    if not (task):
+        print("Invalid PID.")
+        return
+
+    print(
+        "PID: "
+        + str(pid)
+        + "    TASK: "
+        + hex(task.value_())[2:]
+        + "    CPU: "
+        + str(int(task.cpu))
+        + "    COMMAND: "
+        + task.comm.string_().decode("utf-8")
+    )
+
+    if print_full_data:
+        print(
+            "%-5s %-20s %-20s %-20s %-20s %-20s"
+            % (
+                "FD",
+                "SOCKET",
+                "SOCK",
+                "FAMILY:TYPE",
+                "SOURCE-PORT",
+                "DESTINATION-PORT",
+            )
+        )
+
+    rows = [
+        [
+            "FD",
+            "SOCKET",
+            "SOCK",
+            "FAMILY:TYPE",
+            "SOURCE-PORT",
+            "DESTINATION-PORT",
+        ]
+    ]
+
+    for i in range(int(task.files.fdt.max_fds)):
```

Comment:
Drgn has a [helper](https://drgn.readthedocs.io/en/latest/helpers.html#drgn.helpers.linux.fs.for_each_file) for iterating over each open file for a task. Please use that instead.